### DISCUSSION TITLE
Help required in Hover functionality implementation

### DISCUSSION THREAD

**Discussion by GSri30** [eclipse-xtext/xtext#2736] [General]
I'm working on implementing a language server to connect it with my monaco-editor via monaco-languageclient.
I'm trying to implement hover functionality on my server side. The client is already communicating the `textDocument/hover` method rpc message to the server via sockets. 
I couldn't find enough documentation to proceed further. Can anyone help me out.

**Comment by cdietrich** [comment]
Why are you exactly missing

**Reply by GSri30** [reply]
Unlike for validations, there wasn't any templated code that got generated for hover feat.

**Reply by cdietrich** [reply]
There is a default impl. If you don’t like it create and bind your own impl of IHoverService in IdeModule

**Reply by cdietrich** [reply]
https://github.com/xtext/xtext-languageserver-example/blob/80df168e906e6a36b937f698418017c47b30cb70/org.xtext.example.mydsl.ide/src/org/xtext/example/mydsl/ide/MyDslIdeModule.java#L19

**Reply by GSri30** [reply]
Thanks. This helps.

Similarly, can you send the config required for semantic coloring?

**Reply by cdietrich** [reply]
have none at hand place check the tests for it in xtext codebase

**Reply by GSri30** [reply]
![Screenshot 2023-07-11 at 2 30 34 PM](https://github.com/eclipse/xtext/assets/54752266/36e7e9a9-2967-4f12-a6a3-8ac481d16ad9)

I could find this from https://www.eclipse.org/forums/index.php/t/1096951/

But couldn't understand what config I need to use this class.

I'm using gradle.

**Reply by cdietrich** [reply]
this is about eclipse, not about lsp.
org.eclipse.xtext.ide.server.semantictokens.SemanticTokensService.getPositions(XtextResource, CancelIndicator)
simp y bind ISemanticHighlightingCalculator in ide module and you should be fine.
please read dependeny injection in docs

**Reply by GSri30** [reply]
Oh get it. Will try to look into that.

Also, is there any way to enhance the appearance of the hover message? I was trying to use inline HTML, but it wasn't rendering the message itself.

**Reply by cdietrich** [reply]
i assume its markdown?
please read up the spec

**Reply by GSri30** [reply]
Yeah, messages are sent in markdown. But how can I support color specs.

**Reply by cdietrich** [reply]
i dont know.

**Reply by cdietrich** [reply]
is there any other language servers that can?

**Reply by GSri30** [reply]
Ahh, need to check that out.
